src/server/hotspot_partition_calculator.h

```diff
@@ -24,6 +24,9 @@
 namespace pegasus {
 namespace server {
 
+typedef std::list<std::vector<hotspot_partition_data>> statistical_histories;
+typedef std::vector<std::vector<std::unique_ptr<dsn::perf_counter_wrapper>>> hot_partition_counters;
```

Comment:
`std::vector<std::vector<......>>> hot_partition_counters`, This sturcture is a little bit complex, could you please explain why you design it like that?

Reply:
`hot_partition_counters c[index_of_partitions][type_of_read(0)/write(1)_stat]`
so if we have n partitions, we will get 2*n hot_partition_counters, to demonstrate both read/write hotspot value

Reply:
Well, I still think it is complex to read. How about seperate read and write? If there will only have read and write needing to calculate hotspot value, like below:
```
typedef std::vector<std::unique_ptr<dsn::perf_counter_wrapper>> hot_partition_counters;
hot_partition_counters _read_hot_points;
hot_partition_counters _write_hot_points;
```
Besides, could you please also explain `statistical_histories` meaning? It is also complex.

Reply:
If do like that, I will write two duplicate codes to complete one function.
I admit this is a complex structure, but I can do like this to avoid copying codes.
```c++
    for (int data_type = 0; data_type <= 1; data_type++) {
        // 0: READ_HOTSPOT_DATA; 1: WRITE_HOTSPOT_DATA
       ...
        for (const auto &one_partition_stat_histories : _partitions_stat_histories) {
            for (const auto &partition_stat : one_partition_stat_histories) {
                if (partition_stat.total_qps[data_type] > 1.00) {
                    table_qps_sum += partition_stat.total_qps[data_type];
                    ...
                }
            }
        }
```
`statistical_histories` will be renamed into `partitions_stat_histories`, which stores the whole histories of all partitions in one table.